### PR TITLE
fix: update webpack and add BrowserRouter

### PR DESCRIPTION
**Pull request by ajaypratap003**
Changes as required for running federated modules in host app.

1. Updated federatedModuleName as `sr`
2. Added `BrowserRouter` for fixing `Links` issue while running on host app.
3. Added `isProduction` flag and `[chunkhash:8]` to create unique files in prod. It will solve the caching issue.
4. Updated webpack.common.js,  webpack.dev.js and webpack.prod.js files

@EricWittmann Please verify that your `mt-ui` app should work with these changes.

### CODE REVIEW

**Comment by ajaypratap003** [comment]
cc: @pmuir @EricWittmann @paoloantinori 
Please review PR

File: ui/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "repository": "https://github.com/Apicurio/apicurio-registry.git",
   "license": "Apache-2.0",
   "private": true,
-  "federatedModuleName": "apicurio_registry",
+  "federatedModuleName": "sr",
```
Comment (pmuir) [inline ui/package.json:9]: This probably isn't right - instead we should use the `apicurio_registry` name in apps which consume the service registry...

File: ui/webpack.common.js [hunk 2]
```diff
@@ -2,179 +2,198 @@ const path = require("path");
 const TsconfigPathsPlugin = require("tsconfig-paths-webpack-plugin");
 const HtmlWebpackPlugin = require("html-webpack-plugin");
 // webpack 5 stop handling node polyfills by itself, this plugin re-enables the feature
-const NodePolyfillPlugin = require("node-polyfill-webpack-plugin")
```
Comment (pmuir) [inline ui/webpack.common.js:5]: I think this file looks ok, but all the formatting changes make it very hard to see the actual differences

File: ui/webpack.common.js [hunk 3]
```diff
@@ -2,179 +2,198 @@ const path = require("path");
...
+      new HtmlWebpackPlugin({
+        template: "./src/index.html",
+      }),
+      new ChunkMapper({
+        modules: [federatedModuleName],
+      }),
+      new ModuleFederationPlugin({
+        name: federatedModuleName,
+        filename: `${federatedModuleName}${
+          isProduction ? "[chunkhash:8]" : ""
```
Comment (pmuir) [inline ui/webpack.common.js:27]: btw I forgot a dot in this pattern when I did it orginally.
```suggestion
          isProduction ? ".[chunkhash:8]" : ""
```

**Review by pmuir — COMMENTED** [review]
Mostly fine except I'm not sure we want to change the name of the federated module like that, and the webpack.common.js is probably ok but hard to review due to the format changes.

File: ui/package.json (hunk 1 above)
Comment (ajaypratap003) [reply ui/package.json:9]: @pmuir Then we need to do changes in  `config`  for `appication-services-ui` and  `ui-shared`
Should I update `sr`=>`apicurio_registry`  in `appication-services-ui` and  `ui-shared`  ? Please confirm.

File: ui/webpack.common.js (hunk 3 above)
Comment (ajaypratap003) [reply ui/webpack.common.js]: done

**Comment by EricWittmann** [comment]
This all seems reasonable to me.  I'll wait for the final verdict on whether we need to change the federated module name before merging this.

File: ui/package.json (hunk 1 above)
Comment (pmuir) [reply ui/package.json]: Yes sounds good. You'll also need to use it in `FederatedModule` components.

**Comment by ajaypratap003** [comment]
@EricWittmann @pmuir I've revert the module name and put it back ` apircurio_registry`. I'll make changes in host app config and ui-shared. we can merged this PR if everyone ok.

**Review by EricWittmann — APPROVED** [review]

**Pull request merged**